code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -704,17 +706,16 @@ var/global/list/obj/item/device/pda/PDAs = list()
 		return
 
 	if (usr.canUseTopic(src))
-		var/obj/item/weapon/pen/O = locate() in src
-		if(O)
-			if (istype(loc, /mob))
-				var/mob/M = loc
-				if(M.get_active_hand() == null)
-					M.put_in_hands(O)
-					usr << "<span class='notice'>You remove \the [O] from \the [src].</span>"
+		if(is_type_in_list(inserted_item, contained_item))
+			if(inserted_item)
+				if(istype(loc, /mob))
+					var/mob/M = loc
+					M.put_in_hands(inserted_item)
+					usr << "<span class='notice'>You remove \the [inserted_item] from \the [src].</span>"
+					inserted_item = null
 					return
-			O.loc = get_turf(src)
-		else
-			usr << "<span class='warning'>This PDA does not have a pen in it!</span>"
+				inserted_item.forceMove()

```

Comment:
this line is moving the inserted_item into null space, forceMove takes the new location as argument